### PR TITLE
Add image source and alt to HeroSection

### PR DESCRIPTION
**Pull request by newbeelearn**
Added image source and image alt to HeroSection so that images can be
passed to it. Right now it is hardcoded in HeroSection. This will improve
reusability of HeroSection by theme users.

### CODE REVIEW

File: src/components/sections/landing/HeroSection.astro [hunk 1]
```diff
@@ -18,7 +17,9 @@ const {
   avatars,
   starCount,
   rating,
-  reviews
+  reviews,
+  imageSrc,
```
Comment (mearashadowfax) [inline src/components/sections/landing/HeroSection.astro:21]:
```suggestion
  src,
```

File: src/components/sections/landing/HeroSection.astro [hunk 2]
```diff
@@ -75,15 +78,17 @@ interface Props {
...
   <!-- Hero Image Section -->
   <div class="flex w-full">
-    <div class="top-12 overflow-hidden">
-      <Image
-        src={heroImage}
-        alt="Stack of ScrewFast product boxes containing assorted hardware tools"
+      <div class="top-12 overflow-hidden">
+	{imageSrc && imageAlt &&
+	<Image
+        src={imageSrc}
```
Comment (mearashadowfax) [inline src/components/sections/landing/HeroSection.astro:84]:
```suggestion
        src={src}
```

File: src/components/sections/landing/HeroSection.astro [hunk 3]
```diff
@@ -75,15 +78,17 @@ interface Props {
...
+        alt={imageAlt}
```
Comment (mearashadowfax) [inline src/components/sections/landing/HeroSection.astro:85]:
```suggestion
        alt={alt}
```

File: src/components/sections/landing/HeroSection.astro [hunk 4]
```diff
@@ -75,15 +78,17 @@ interface Props {
...
         class="h-full w-full scale-110 object-cover object-center"
         draggable={"false"}
         loading={"eager"}
-        format={"avif"}
-      />
+	 />
```
Comment (mearashadowfax) [inline src/components/sections/landing/HeroSection.astro:89]: Also adding back `format={"avif"}` for our case. By default, the <Image /> component will produce a .webp file.

File: src/pages/index.astro [hunk 5]
```diff
@@ -39,6 +40,8 @@ const avatarSrcs: Array<string> = [
     rating=`<span class="font-bold">4.8</span> / 5`
     starCount={4}
     reviews=`From Over <span class="font-bold">12.8k</span> Reviews`
+    imageSrc={heroImage}
```
Comment (mearashadowfax) [inline src/pages/index.astro:43]:
```suggestion
    src={heroImage}
```

File: src/pages/index.astro [hunk 6]
```diff
@@ -39,6 +40,8 @@ const avatarSrcs: Array<string> = [
...
+    imageAlt="Stack of ScrewFast product boxes containing assorted hardware tools"
```
Comment (mearashadowfax) [inline src/pages/index.astro:44]:
```suggestion
    alt="Stack of ScrewFast product boxes containing assorted hardware tools"
```

File: src/components/sections/landing/HeroSection.astro (hunk 4 above)
Comment (mearashadowfax) [reply src/components/sections/landing/HeroSection.astro:89]: In my opinion, we can just keep it in the Image component of a Hero section as it was.

**Comment by newbeelearn** [comment]
i will make similar changes for feature and featurenavs as well. I think astro.build did not thought of the usecase where themes can be updated independently while themeusers continue to use it so they get all the bug fixes/improvements without having to do the confilct resolution everytime parent theme is updated. So now that burden will either fall on theme creator or theme user. Ideally users should be able to override theme without modifying it hugo ssg does it by providing users a way to override theme by having separate directory for user layouts.

File: src/components/sections/landing/HeroSection.astro (hunk 4 above)
Comment (newbeelearn) [reply src/components/sections/landing/HeroSection.astro:89]: ok

**Comment by mearashadowfax** [comment]
> i will make similar changes for feature and featurenavs as well. 

I recall that in those component blocks, I made the images dynamic.

**Comment by newbeelearn** [comment]
I will create a new pull request for this change with all the changes that were suggested. Closing the current request.

**Pull request closed without merging**